sdk/core/core/CMakeLists.txt

```diff
@@ -7,9 +7,22 @@ project (az_core LANGUAGES C)
 
 set(CMAKE_C_STANDARD 99)
 
-add_library (az_core src/az_core.c)
+add_library (az_core src/az_core.c src/az_json_read.c)
+
+if(MSVC)
+  target_compile_options(az_core PRIVATE /W4 /WX)
+else()
+  target_compile_options(az_core PRIVATE -Wall -Wextra -pedantic -Werror)
+endif()
 
 target_include_directories (az_core PUBLIC inc)
 
 # make sure that users can consume the project as a library.
 add_library (az::core ALIAS az_core)
+
+add_executable (az_core_test test/main.c)
+
+target_include_directories (az_core_test PRIVATE inc)
+target_link_libraries(az_core_test PRIVATE az_core)
+
+add_test(NAME az_core_test COMMAND ${EXECUTABLE_OUTPUT_PATH}/az_core_test)
```

Comment:
you can just write `add_test(NAME az_core_test COMMAND az_core_test)` if COMMAND is a target name cmake will figure out the required path on its own.